src/controllers/messagesController.ts

```diff
@@ -39,8 +39,8 @@ export default class MessagesController {
       withValidatedOrValidationError(
         GetMessagesParameters.decode({
           /* eslint-disable sort-keys */
-          pageSize: req.query.page_size,
-          enrichResultData: req.query.enrich_result_data,
+          pageSize: parseInt(req.query.page_size, 10),
+          enrichResultData: req.query.enrich_result_data === "true",
```

Comment:
I wonder if that would have been better to have a io-ts type which defines

`pageSize` as `NonNegativeIntegerFromString` and `enrichResultData` as `BooleanFromString`
before the call to `decode()`

https://github.com/pagopa/io-ts-commons/blob/master/src/numbers.ts#L124
https://github.com/pagopa/io-ts-commons/blob/master/src/booleans.ts

Reply:
surely better!
I didn't know we have also these codecs
Changed in d9ac9937623bb184833cba68c8b98a5d3ebb3d9a